daemon-tests/tests/happy_path.rs

```diff
@@ -427,6 +427,66 @@ async fn maker_rejects_rollover_after_commit_finality() {
     wait_next_state!(order_id, maker, taker, CfdState::OpenCommitted);
 }
 
+#[tokio::test]
+async fn maker_rejects_collab_settlement_after_commit_finality() {
+    let _guard = init_tracing();
+    let (mut maker, mut taker, order_id) =
+        start_from_open_cfd_state(OliviaData::example_0().announcement(), Position::Short).await;
+    taker.mocks.mock_latest_quote(Some(dummy_quote())).await;
+    maker.mocks.mock_latest_quote(Some(dummy_quote())).await;
+    next_with(taker.quote_feed(), |q| q).await.unwrap(); // if quote is available on feed, it propagated through the system
+
+    taker.system.propose_settlement(order_id).await.unwrap();
+
+    wait_next_state!(
+        order_id,
+        maker,
+        taker,
+        CfdState::IncomingSettlementProposal,
+        CfdState::OutgoingSettlementProposal
+    );
+
+    confirm!(commit transaction, order_id, maker, taker);
+    // Cfd would be in "OpenCommitted" if it wasn't for the rollover
+
+    maker.system.reject_settlement(order_id).await.unwrap();
+    sleep(Duration::from_secs(5)).await; // need to wait a bit until both transition
+
+    // After rejecting rollover, we should display where we were before the
+    // rollover attempt
+    wait_next_state!(order_id, maker, taker, CfdState::OpenCommitted);
+}
+
+#[tokio::test]
+async fn maker_accepts_collab_settlement_after_commit_finality() {
```

Comment:
I think it would help if we have longer test names:

suggestions:

`given_maker_committed_when_accept_collab_settlement_then_remains_committed`
`given_maker_committed_when_rejects_collab_settlement_then_remains_committed`

It took me some time to understand why these tests are there and what is the difference.

Reply:
I actually would have harder time parsing the suggested names. :/